### ISSUE TITLE
Transition to GitHub-native auto-merge feature

### ISSUE THREAD

**Issue by mitchdenny** [Azure/azure-sdk-tools#1289]
Currently we make use of fabric bot to auto-merge PRs which have the auto merge label once all the checks have passed in the PR is approved. GitHub recently rolled out a native automerge feature so we should plan on moving to that. This work will involve:

1. Try out automerge in a repo and make sure it works as advertised.
2. Change automation that creates PRs with automerge labels (e.g. eng common sync) to set auto merge flag instead.
3. Disable the fabric bot rules.

It isn't clear at this point whether automerge settings are exposed on the REST API. This would probably be the initial investigation.

**Comment by sima-zhu** [comment]
https://support.github.com/ticket/personal/0/982959

**Comment by weshaggard** [comment]
I've been testing some of this in the azure-sdk repo at https://github.com/Azure/azure-sdk/blob/main/eng/pipelines/version-updater.yml#L91-L97

**Comment by weshaggard** [comment]
Closing in favor of https://github.com/Azure/azure-sdk/issues/5832
